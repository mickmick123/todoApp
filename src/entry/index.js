import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import MainNavigation from '../navigators/mainNavigation';

const Entry = () => {
  return (
    <NavigationContainer>
      <MainNavigation />
    </NavigationContainer>
  );
};

export default Entry;
