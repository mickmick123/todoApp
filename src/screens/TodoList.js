import React from 'react';
import Container from '../components/Container';
import Header from '../components/Header';
import {Screen} from '../components/Screen';
import {useDispatch, useSelector} from 'react-redux';
import ListItems from '../components/ListItems';
import {deleteTask, updateTasksCompleted} from '../store/actions/tasks';
import styles from '../styles/base';
import {View, Text} from 'react-native';

const TodoList = ({navigation}) => {
  const {tasks} = useSelector(state => state.todo);
  const dispatch = useDispatch();

  const deleteRow = rowKey => {
    dispatch(deleteTask(rowKey));
  };

  const moveToComplete = rowKey => {
    const isolatedData = tasks.find(item => item.key === rowKey);
    dispatch(updateTasksCompleted(isolatedData));
    deleteRow(rowKey);
  };

  return (
    <Container>
      <Header
        title="Todo Lists"
        onPress={() => navigation.navigate('AddTask')}
        icon="pluscircle"
        hasTab={true}
        canSort={true}
      />
      <Screen preset="fixed" safeAreaEdges={['top', 'bottom']}>
        {tasks && tasks.length ? (
          <ListItems
            listItem={tasks}
            deleteRow={deleteRow}
            moveToComplete={moveToComplete}
            isActiveList={true}
          />
        ) : (
          <View style={styles.defaultList}>
            <Text style={styles.defaultListText}>No Active Tasks</Text>
          </View>
        )}
      </Screen>
    </Container>
  );
};

export default TodoList;
