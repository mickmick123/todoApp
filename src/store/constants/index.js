export const FILTER_UPDATE = 'FILTER_UPDATE';
export const TASK_UPDATE_COMPLETED = 'TASK_UPDATE_COMPLETED';
export const ADD_TASK = 'ADD_TASK';
export const DELETE_TASK = 'DELETE_TASK';
export const DELETE_COMPLETED_TASK = 'DELETE_COMPLETED_TASK';
