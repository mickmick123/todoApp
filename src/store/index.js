import {createStore, combineReducers} from 'redux';
import taskReducer from './reducers/taskReducer';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {persistStore, persistReducer} from 'redux-persist';

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
};

const rootReducer = combineReducers({
  todo: taskReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = createStore(persistedReducer);
export const persistor = persistStore(store);
