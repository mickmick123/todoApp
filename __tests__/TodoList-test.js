import React from 'react';
import {Provider} from 'react-redux';
import {persistor, store} from '../src/store';
import {PersistGate} from 'redux-persist/integration/react';
import {SafeAreaProvider} from 'react-native-safe-area-context';
import {render, fireEvent, cleanup} from '@testing-library/react-native';
import {TouchableOpacity} from 'react-native';

import TodoList from '../src/screens/TodoList';

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.mock('@react-native-async-storage/async-storage', () =>
  require('@react-native-async-storage/async-storage/jest/async-storage-mock'),
);
jest.mock('@react-navigation/bottom-tabs');
jest.mock('@react-navigation/native-stack');
jest.mock('react-native-vector-icons/AntDesign', () => 'Icon');
jest.mock('react-native-vector-icons/Ionicons', () => 'Ionicons');
jest.mock('react-native-vector-icons/FontAwesome5', () => 'FontAwesome5');
jest.mock('react-native-form-validator', () => 'useValidation');

jest.mock('redux-persist', () => {
  const real = jest.requireActual('redux-persist');
  return {
    ...real,
    persistReducer: jest
      .fn()
      .mockImplementation((config, reducers) => reducers),
  };
});
jest.mock('redux-persist/integration/react', () => ({
  PersistGate: props => props.children,
}));
const navigation = {
  navigate: jest.fn(),
  goBack: jest.fn(),
};

const tree = render(
  <Provider store={store}>
    <PersistGate persistor={persistor}>
      <SafeAreaProvider>
        <TodoList navigation={navigation} />
      </SafeAreaProvider>
    </PersistGate>
  </Provider>,
);

describe('Test todo list page', () => {
  afterEach(cleanup);
  it('page is render', () => {
    expect(tree).toMatchSnapshot();
  });
  it('navigate to add task', async () => {
    const {findByTestId} = render(
      <TouchableOpacity
        testID="button"
        onPress={() => navigation.navigate('AddTask')}
      />,
    );
    const button = await findByTestId('button');
    fireEvent.press(button);
    expect(navigation.navigate).toBeCalledWith('AddTask');
  });
});
