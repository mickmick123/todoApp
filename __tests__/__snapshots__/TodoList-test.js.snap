// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test todo list page page is render 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
